bindings/java/src/org/sleuthkit/datamodel/SleuthkitCase.java

```diff
@@ -131,8 +132,23 @@
 
 	private CommunicationsManager communicationsMgrInstance;
 	private TimelineManager timelineMgrInstance;
+	private Blackboard blackboardInstance;
 
-	private final Map<String, Set<Long>> deviceIdToDatasourceObjIdMap = new HashMap<String, Set<Long>>();
+	private final Map<String, Set<Long>> deviceIdToDatasourceObjIdMap = new HashMap<>();
+
+	private final EventBus eventBus = new EventBus("SleuthkitCase-EventBus");
+
+	public void registerForEvents(Object listener) {
+		eventBus.register(listener);
+	}
+
+	public void unregisterForEvents(Object listener) {
+		eventBus.unregister(listener);
+	}
+
+	void postTSKEvent(Object event) {
+		eventBus.post(event);
+	}
```

Comment:
@bcarrier and @millmanorama I believe we should be moving towards simplifying SleuthkitCase so that one day it provides database primitives to "manager" classes that partition its functionality, i.e., the direction we are heading with CommunicationsManager, TimelineManager, Blackboard, etc. Putting a SleuthKit layer event bus in this class therefore seems like a step in the wrong direction. I think that we need a separate SleuthkitEventBus abstraction for the event bus. A mirroring API consisting of register, unregister and post would make sense to me. Perhaps, if we need to tie the EventBus to given case, the reference could be obtained through SleuthkitCase like the "managers" are, but I would not make it part of SleuthkitCase.

Reply:
I am fine with this if this is way we want to go.  It should not be that hard.